### ISSUE TITLE
window:update_win_title can cause invalid UTF-8

### ISSUE THREAD

**Issue by Kyuuhachi** [luakit/luakit#673]
Since Lua's strings deal with bytes rather than characters, they can sometimes cut off halfway through a character. This makes numerous programs fail or crash, including `i3` and `xwininfo`/`xprop`.

One example of a failing website is [this](https://www.fanfiction.net/s/10552390/45/Magic-Online), with `window.max_title_len` at the default of 80.

**Comment by Kyuuhachi** [comment]
Can be worked around by setting `window.max_title_len` to a copious amount, or by overwriting `update_win_title` with another function such as this:

```lua
window.methods.update_win_title = function (w)
	local uri, title = w.view.uri, w.view.title
	w.win.title = (title or "…") .. ((uri and " - " .. uri) or "")
end
```

**Comment by aidanholm** [comment]
Thanks for reporting! Fixed at 877e25c0
